TESTING.md

```diff
@@ -1,6 +1,153 @@
-# VSCode testing
+# ACPaaS UI Testing Guide
 
-To make life easier in VScode for testing with Jest.
-We suggest you install the following plugin:
+When developing, testing is a major part of the work.
+We use [Jest](https://jestjs.io) in combination with [Enzyme](https://airbnb.io/enzyme) as our test suite.
 
-https://github.com/jest-community/vscode-jest
\ No newline at end of file
+## VSCode testing
+
+To make life easier in Visual Studio Code for testing with Jest, we suggest that you install [this VS Code Jest extension](https://github.com/jest-community/vscode-jest).
+
+## ACPaaS UI Test Automation Guide
+
+When automating tests, one of the challenges is to keep the test resilient to code changes.
+Using [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) can help you with that, by creating unique selectors for your tests.
+
+### Using unique data attributes with ACPaaS UI
+
+It's not possible to add unique data attributes to the ACPaaS UI component library for the simple reason that the attribute won't be unique anymore when you use the component multiple times on the same page.
+
+But in your app you can simply add a data attribute to each individual ACPaaS UI component so you can target the component that you want to test.
+
+#### Example
+
+The following example shows the ACPaaS UI Accordion with the unique `qa` data attribute `id-1234`.
+
+```html
+<Accordion qa="id-1234"/>
+    <AccordionTab />
+    <AccordionTab />
+</Accordion>
+```
+
+Which will result in:
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  ...
+</div>
+```
+
+### Branding classes
+
+Since you probably want to target a specific element or a subcomponent, selecting the parent component won't be enough.
+You can however target these elements and subcomponents based on their CSS classes in combination with the unique data attribute of the parent component.
+
+#### Example
+
+In this example the accordiontab can be opened or closed by toggling the `<div>` tag with the CSS class `m-accordion__tab`, which is nested in the accordion with data attribute `data-qa="id-1234"`.
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  <div ... class="m-accordion__tab">
+  ...
+</div>
+```
```

Comment:
Maybe it's useful to mention that this is compiled code.

Reply:
Done.